### PR TITLE
Update PesterTests and devcontainer to .NET 8

### PR DESCRIPTION
**Pull request by fflaten**
## PR Summary
.NET 6 is reaching EOL later this year. PR changes:
- Updates devcontainer image to .NET 8 SDK. Adds extensions to improve pipelines/workflows DX
- Updates PesterTests to .NET 8 runtime and latest dependencies
- Updates CONTRIBUTING docs to require .NET 8 SDK
- Ignores build warning about vulnerable PowerShell SDK dependency (<6.1.6), related #2207

.NET 8 SDK is already used for building in Azure Pipelines CI.

## PR Checklist

- [x] PR has meaningful title
- [x] Summary describes changes
- [x] PR is ready to be merged
  - If not, use the arrow next to `Create Pull Request` to mark it as a draft. PR can be marked `Ready for review` when it's ready.
- [ ] Tests are added/update *(if required)*
- [ ] Documentation is updated/added *(if required)*

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -53,9 +53,9 @@ Get-Help ./build.ps1 -Detailed
 
 Pester has a C# Solution which requires .NET Framework SDKs and Developer Packs in order to compile. The targeted frameworks can be found in `src\csharp\Pester\Pester.csproj`.
 
-### Install .NET Core 6.0 SDK
+### Install .NET 8.0 SDK
 
-[Download Link](https://dotnet.microsoft.com/download/dotnet-core/6.0)
+[Download Link](https://dotnet.microsoft.com/en-us/download/dotnet/8.0)
 
 ### .Net Framework 4.5 Developer Pack
```
Comment (fflaten) [inline CONTRIBUTING.md:60]: Can this be removed? Necessary now that we use `Microsoft.NETFramework.ReferenceAssemblies` package in assembly?

Comment (nohwnd) [reply CONTRIBUTING.md:60]: Probably not (I have all of them installed so I cannot easily check), but the whole point of that nuget is that you can build .NET Framework even on Linux where it cannot run (without mono).

Comment (nohwnd) [inline CONTRIBUTING.md:60]:
```suggestion
```

Comment (nohwnd) [reply CONTRIBUTING.md:60]: Cannot make suggestion that deletes it, please delete it.

**Comment by nohwnd** [comment]
Partly addressed in #2432, I will merge the changes once it is in.

**Pull request merged**